crates/primitives/src/header.rs

```diff
@@ -284,9 +284,9 @@ impl Decodable for Header {
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct SealedHeader {
     /// Locked Header fields.
-    header: Header,
+    pub header: Header,
```

Comment:
Why mark this as pub? `SealedHeader` implements `AsRef` and `Deref` where the target is `Header`, and that is how we use it all over the code base. If we mark this pub I think we should remove those traits and the `hash()` function.

Reply:
Same reason why i made `hash` pup. I think adding `AsMut` and `DerefMut` is a better way.

Reply:
Reread this. `Deref` allows us to have `block: SealedBlock` but access number with `block.number`, if we didn't have it it would be `block.block.header.header.number` and that is noisy.
Removing `hash()` is just search and replace but it will just make a noise in the PR. Will do it after the merge.